### PR TITLE
close #1170 Notification tap does navigate to the chat on Android when app is not started

### LINKED ISSUE #1170: Notification tap does navigate to the chat on Android when app is not started

### PR DESCRIPTION
**Pull request by crui3er**

### CODE REVIEW

File: src/dotnet/Notification.UI.Blazor/NotificationUI.cs [hunk 1]
```diff
@@ -94,23 +96,17 @@ public Task RequestNotificationPermissions(CancellationToken cancellationToken)
...
-            try {
-                await HandleNotificationNavigation(url);
-            }
-            catch (Exception e) {
-                Log.LogError(e, "Failed to dispatch notification navigation");
-            }
-        });
+        => _ = Dispatcher.CheckAccess()
+            ? HandleNotificationNavigation(url)
+            : Dispatcher.InvokeAsync(async () => await HandleNotificationNavigation(url));
```
Comment (frolyo) [inline src/dotnet/Notification.UI.Blazor/NotificationUI.cs:101]:
```suggestion
            : Dispatcher.InvokeAsync(() => HandleNotificationNavigation(url));
```

File: src/dotnet/UI.Blazor/Services/NavigationCoordinatorUI.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        // including account is loaded and initial redirect to '/chat/'
+        await WhenReady.ConfigureAwait(true);
+        await History.NavigateTo(localUrl).ConfigureAwait(true);
+
+        using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3));
+        await History
+            .When(_ => History.LocalUrl.Value.OrdinalStartsWith(localUrl), cts.Token)
+            .SuppressCancellation().ConfigureAwait(true);
+        await History.WhenNavigationCompleted.ConfigureAwait(true);
+        _outstandingRequestsNumber--;
```
Comment (frolyo) [inline src/dotnet/UI.Blazor/Services/NavigationCoordinatorUI.cs:43]: is code above 100% reliable?

**Review by frolyo — APPROVED** [review]

File: src/dotnet/UI.Blazor/Services/NavigationCoordinatorUI.cs (hunk 2 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor/Services/NavigationCoordinatorUI.cs:43]: What do you mean? What do you concern about?

Comment (frolyo) [reply src/dotnet/UI.Blazor/Services/NavigationCoordinatorUI.cs:43]: that on error you will have hanging outstanding requests

**Pull request merged**